src/reader/ArticleReader.js

```diff
@@ -90,9 +79,19 @@ export default function ArticleReader({ api, teacherArticleID }) {
     );
   }
 
+  function updateScrollPosition() {
+    var scrollElement = document.getElementById("scrollHolder");
+    var scrollY = scrollElement.scrollTop;
+    var limit = scrollElement.scrollHeight - scrollElement.clientHeight - 450; // 450 represents the feedback + exercise div
+    var ratio = Math.round((scrollY / limit) * 100) / 100;
```

Comment:
opportunity for `ratio` utility function?

Reply:
Done!